### DISCUSSION TITLE
Entering room with #waypoint URL works, reload does not

### DISCUSSION THREAD

**Discussion by blairmacintyre** [mozilla/hubs#5145] [Hubs Cloud Questions]
The current hubs cloud client supports entering a room with the #waypoint at the end of the URL.

BUT, when I hit refresh on the browser, the room reloads at the default spawn point (or one of the default spawns), since the # has been removed.

I'm not entirely sure of the best way to fix this, but I do wonder why hubs mucks about with the browser history so much.  Why not just leave the URLs as they are, so that things like refresh, bookmarking, copying URLs, etc. work?

**Comment by jywarren** [comment]
In some systems, like this minecraft tilemap generator, the URL hash is updated regularly to reflect the current location in-world, either by coordinates (example.com#z/x/y) or by proximity to named locations (like the waypoints here). This could be one way to "respawn back where you were" or close to it. If you pass by another waypoint, it could be "sticky" so you spawn near the last waypoint you passed, kind of like when resuming a FPS game. 

https://github.com/overviewer/Minecraft-Overviewer/pull/348 has the implementation I wrote years ago, though some browser behaviors may have changed!
